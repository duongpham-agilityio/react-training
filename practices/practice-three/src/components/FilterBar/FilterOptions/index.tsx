import { memo } from 'react';
import { Button, Link, List, ListItem } from '@chakra-ui/react';

// Mocks
import { filterOptions } from '@/mocks';

export const FilterOptions = memo(() => (
  <List display="flex" gap={5} py={5}>
    {filterOptions.map((option) => {
      const { name, value } = option;

      return (
        <ListItem key={value}>
          <Button
            as={Link} // Todo: Update to later
            href={`?filter_option=${value}`} // Todo: Update to later
            border="1px solid"
            borderColor="blackAlpha.300"
            borderRadius={25}
            boxShadow="md"
            bg="white"
            color="gray.20"
            fontWeight="regular"
            fontSize={16}
            px={15}
            _hover={{
              boxShadow: 'xl',
            }}
          >
            {name}
          </Button>
        </ListItem>
      );
    })}
  </List>
));
