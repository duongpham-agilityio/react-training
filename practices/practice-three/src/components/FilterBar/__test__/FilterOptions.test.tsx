import { render } from '@testing-library/react';

// Components
import { FilterOptions } from '../FilterOptions';

const setup = () => render(<FilterOptions />);

describe('Filteroptions', () => {
  it('Match to snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
