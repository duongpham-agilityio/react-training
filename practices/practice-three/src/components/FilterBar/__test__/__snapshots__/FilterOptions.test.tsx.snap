// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filteroptions Match to snapshot 1`] = `
<div>
  <ul
    class="css-1c8ne0r"
    role="list"
  >
    <li
      class="css-0"
    >
      <a
        class="chakra-link chakra-button css-99bvj6"
        href="all"
      >
        All
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        class="chakra-link chakra-button css-99bvj6"
        href="fashion"
      >
        Fashion
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        class="chakra-link chakra-button css-99bvj6"
        href="electrics"
      >
        Electrics
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        class="chakra-link chakra-button css-99bvj6"
        href="books"
      >
        Books
      </a>
    </li>
  </ul>
</div>
`;
